types/types.go

```diff
@@ -0,0 +1,141 @@
+package types
+
+type Block struct {
+	BlockHash       string         `json:"block_hash"`
+	ParentBlockHash string         `json:"parent_hash"`
+	BlockNumber     int            `json:"block_number"`
+	NewRoot         string         `json:"new_root"`
+	OldRoot         string         `json:"old_root"`
+	Status          string         `json:"status"`
+	AcceptedTime    uint64         `json:"accepted_time"`
+	Transactions    []*Transaction `json:"transactions"`
+}
+
+type Code struct {
+	Bytecode []string `json:"bytecode"`
+	Abi      []struct {
+		Inputs []struct {
+			Name string `json:"name"`
+			Type string `json:"type"`
+		} `json:"inputs"`
+		Name            string        `json:"name"`
+		Outputs         []interface{} `json:"outputs"`
+		Type            string        `json:"type"`
+		StateMutability string        `json:"stateMutability,omitempty"`
+	} `json:"abi"`
+}
+
+/*
+	StarkNet transaction states
+*/
+const (
+	NOT_RECIEVED = TxStatus(iota)
+	REJECTED
+	RECEIVED
+	PENDING
+	ACCEPTED_ON_L2
+	ACCEPTED_ON_L1
+)
+
+var TxStatuses = []string{"NOT_RECEIVED", "REJECTED", "RECEIVED", "PENDING", "ACCEPTED_ON_L2", "ACCEPTED_ON_L1"}
+
+type TxStatus int
+
+func (s TxStatus) String() string {
+	return TxStatuses[s]
+}
+
+type TransactionStatus struct {
+	TxStatus        string `json:"tx_status"`
+	BlockHash       string `json:"block_hash"`
+	TxFailureReason struct {
+		ErrorMessage string `json:"error_message,omitempty"`
+	} `json:"tx_failure_reason,omitempty"`
+}
+
+type Transaction struct {
+	TransactionReceipt
```

Comment:
Hm it is to support the `FULL_TXN_AND_RECEIPTS` scope, which returns the transactions with receipts inlined. Otherwise we'll need to introduce a new type to handle the marshaling. I can take a look at `signer.Execute` but lmk if you have any other ideas

Reply:
cool i updated the structs to be pointers and to omit them if nil. looks like the tests are passing now!